pallets/loans/src/types.rs

```diff
@@ -138,19 +116,156 @@ pub enum NAVUpdateType {
+// 	End,
+// }
+
+// pub enum InterestReferenceDate {
+// 	/// Interest payments are expected every period based on an event, e.g. the end of the period
+// 	/// E.g. if the period is monthly and the origination date is Mar 3, the first interest
+// 	/// payment is expected on Mar 31.
+// 	CalendarDate { event: CalendarEvent },
+// 	/// Interest payments are expected every period relative to the origination date.
+// 	/// E.g. if the period is monthly and the origination date is Mar 3, the first interest
+// 	/// payment is expected on Apr 3.
+// 	OriginationDate,
+// }
+
+// TODO: implement Yearly, Quarterly, Daily
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum InterestPayments {
+	/// All interest is expected to be paid at the maturity date
+	None,
+	// Interest payments are expected monthly
+	// Monthly { reference: InterestReferenceDate },
+}
+
+// TODO: implement StraightLine, Annuity
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum PayDownSchedule {
+	/// The entire borrowed amount is expected to be paid back at the maturity date
+	None,
+}
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Maturity {
+	Fixed(Moment),
+}
 
-	// interest rate per second
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct RepaymentSchedule {
+	/// Expected repayment date for remaining debt
+	maturity: Maturity,
+	/// Period at which interest is paid
+	interest_payments: InterestPayments,
+	/// How much of the initially borrowed amount is paid back during interest payments
+	pay_down_schedule: PayDownSchedule,
+}
+
+impl RepaymentSchedule {
+	pub fn maturity_date(&self) -> Moment {
+		match self.maturity {
+			Maturity::Fixed(moment) => moment,
+		}
+	}
+
+	pub fn update_maturity_date(&mut self, date: Moment) {
+		match &mut self.maturity {
+			Maturity::Fixed(moment) => *moment = date,
+		}
+	}
+}
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum BorrowRestrictions {
+	None,
+}
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum RepayRestrictions {
+	None,
+}
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum PriceRestrictions {
+	None,
+}
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum MaxBorrowAmount<Rate> {
+	/// Collateral value - advance rate * total borrowed
```

Comment:
From ceiling computation [here](https://centrifuge.hackmd.io/pYxf6rE-S2yYApZZpkGC1w), and what is done currently, I understand that the comment should be:
```suggestion
	/// Collateral value * advance rate - total borrowed
```

Reply:
Can I have a double-check of this?

Reply:
Yeah you're right!